nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -45,65 +45,153 @@ def _setup_tokenizer(self, tokenizer_cfg: DictConfig):
             os.environ['TOKENIZERS_PARALLELISM'] = 'false'
 
         self.tokenizer_cfg = OmegaConf.to_container(tokenizer_cfg, resolve=True)  # type: dict
-        self.tokenizer_dir = self.tokenizer_cfg.pop('dir')  # Remove tokenizer directory
-        self.tokenizer_type = self.tokenizer_cfg.pop('type').lower()  # Remove tokenizer_type
 
-        self.hf_tokenizer_kwargs = self.tokenizer_cfg.pop("hf_kwargs", {})  # Remove HF tokenizer kwargs
+        # the aggregate tokenizer does not have one tokenizer_dir but multiple ones
+        self.tokenizer_dir = None
+
+        # not popping these variables yet because we need them for processing
+        if 'dir' in self.tokenizer_cfg:
+            self.tokenizer_dir = self.tokenizer_cfg.get('dir')
+
+        self.tokenizer_type = self.tokenizer_cfg.get('type').lower()
+        self.hf_tokenizer_kwargs = self.tokenizer_cfg.get('hf_kwargs', {})
 
         # Preserve config
         if hasattr(self, 'cfg') and 'tokenizer' in self.cfg:
-            self.cfg.tokenizer.dir = self.tokenizer_dir
-            self.cfg.tokenizer.type = self.tokenizer_type
+            with open_dict(self.cfg):
+                if self.tokenizer_dir is not None:
+                    self.cfg.tokenizer.dir = self.tokenizer_dir
+                else:
+                    self.cfg.tokenizer.pop('dir', None)
+                self.cfg.tokenizer.type = self.tokenizer_type
 
             if 'hf_kwargs' in tokenizer_cfg:
                 with open_dict(self.cfg.tokenizer):
                     self.cfg.tokenizer.hf_kwargs = tokenizer_cfg.get('hf_kwargs')
 
-        if self.tokenizer_type not in ['bpe', 'wpe']:
+        if self.tokenizer_type not in ['bpe', 'wpe', 'agg']:
             raise ValueError(
-                "`tokenizer.type` must be either `bpe` for SentencePiece tokenizer or "
+                "`tokenizer.type` must be either `agg` for aggregate, `bpe` for SentencePiece tokenizer or "
                 "`wpe` for BERT based tokenizer"
             )
 
-        if self.tokenizer_type == 'bpe':
+        if self.tokenizer_type == 'agg':
+            logging.debug('_setup_tokenizer: detected an aggregate tokenizer')
```

Comment:
Remove or I think better make it info. To let users know agg tokenizer is built.

Reply:
done!